### ISSUE TITLE
 CohereForAI / c4ai-command-r-plus-4bit

### ISSUE THREAD

**Issue by petunder** [ollama/ollama#3532]
### What model would you like?

C4AI Command R+ is an open weights research release of a 104B billion parameter model with highly advanced capabilities, this includes Retrieval Augmented Generation (RAG) and tool use to automate sophisticated tasks.

https://huggingface.co/CohereForAI/c4ai-command-r-plus-4bit

This model is split into individual files, which makes direct uploading difficult. Additionally, the model is in the .safetensors format.

**Comment by jason-c-kwan** [comment]
@petunder Do instructions exist somewhere on how to build ollama with a different commit/PR of llama.cpp? I have not been able to get it to work so far.

**Comment by petunder** [comment]
> @petunder Do instructions exist somewhere on how to build ollama with a different commit/PR of llama.cpp? I have not been able to get it to work so far.

Two hours ago @ggerganov updated llama.cpp code, so you can build ollama from source as usual, all things should work

**Comment by taozhiyuai** [comment]
support! I need command r plus on ollama. :_)

**Comment by jason-c-kwan** [comment]
@petunder Building does not work for me. Doing `go build .` gives me this error:

```
# github.com/ollama/ollama/llm
llm/server.go:68:28: undefined: gpu.CheckVRAM
llm/server.go:69:14: undefined: gpu.GetGPUInfo
```

...and the executables are not made. Am I missing something obvious here or should I submit a separate issue?

**Comment by jason-c-kwan** [comment]
Oh, I figured it out. For anyone else who comes across the same issue - the problem was I was building in a conda environment. Installing go 1.22 on the base system made it work.

**Comment by trinque** [comment]
https://github.com/ggerganov/llama.cpp/pull/6491 has been merged.

**Comment by jason-c-kwan** [comment]
Just opened a new issue (#3563 ) because I figured this one is specifically for requesting the 4-bit version. I have not been able to import either version of the GGUFs I could find on Hugging Face and not all the quants are yet available on ollama hub.

**Comment by pdevine** [comment]
Let's cover this in #3576 . The changes are already merged and should be available in `0.1.32`.
